torch/csrc/jit/backends/backend_detail.h

```diff
@@ -11,7 +12,7 @@ namespace jit {
 namespace detail {
 
 using BackendPreprocessFunction =
-    std::function<c10::IValue(const Module&, const c10::Dict<IValue, IValue>&)>;
+    std::function<c10::IValue(const Module&, const c10::Dict<IValue, IValue>&, BackendDebugInfoRecorder&)>;
```

Comment:
Since there's already an argument of `c10::Dict<IValue, IValue>&`, why not we wrap the BackendDebugInfoRecorder into an IValue (which can be anything), and put it into the dict, like `{"BackendDebugInfoRecorder": recorder_obj}`? 

By doing this we don't need to change the signature of `BackendPreprocessFunction`. I think it makes sense to put it into the dict as well, as a "method_compile_spec", or "method_compile_util".

Reply:
As far as I understand you cannot wrap any arbitrary cpp object in IValue. For arbitrary cpp objects you have to create custom class out of it first. But if it is possible that would be certainly better.

Reply:
Yeah I think that's right, plus it's nice to have it clearly tagged as something that is core to the preprocessing step.

My suggestion though is to not pass the BackendDebugInfoRecorder directly, but have a simple interface that only focus is to define the ```generate_debug_handles()```.

So, more concretely is to have something along the lines of:

```
class DebugHandleManager { // better names appreciated
  virtual NodeToDebugHandle generate_debug_handles(const std::shared_ptr<Graph>& graph);
}

class BackendDebugInfoRecorder {
  ...
  DebugHandleManager get_debug_handle_manager();
}
```

Basically an interface that is isolate and only focuses on generating the debug handles, so the BackendPreprocessFunction is cleaner.

Reply:
So I understand the question about cleaner interface that does simply debug handle generation. But I would prefer to not introduce yet another class that does just that. Maybe we can just have:
```
BackendDebugInfoRecorder debug_info_recorder;
auto generate_debug_handles_fn = [&debug_info_recorder](const std::shared_ptr<Graph>& g) {
  return debug_info_recorder.generate_debug_handles(g);
  // or 
 return generate_debug_handles(debug_info_recorder, g);
};
loweredModule.register_attribute("__processed_module", AnyType::get(),    
  detail::getBackendPreprocessFunction(backend_name)( cloned_module, method_compile_spec, 
    generate_debug_handles_fn), 
  /*is_param=*/false);
```

What do you think?

Reply:
Yup, that sound good.